### PR TITLE
DSi_SD: implement command 17, 24

### PR DESCRIPTION
**Pull request by asiekierka**
Read/write single block; this will be used by BlocksDS-based homebrew in the near future.

### CODE REVIEW

**Comment by RSDuck** [comment]
is it right that to completely take over the behaviour of the block reads in regards to CSR and RWCommand?

**Comment by asiekierka** [comment]
Not at all, but it did just so happen to work. Let's improve it...

**Comment by asiekierka** [comment]
RWCommand was wrong, but I *think* CSR is actually fine; the data block is sent after a response for both single and multiple reads.

I'd like to test this further, but at some point after 0.9.5, my trick of using a dummy NAND filled with zeroes stopped working (I don't own a proper DSi, only a 3DS - I'd like to look into implementing some code to not require a BIOS/firmware/NAND dump for some forms of DSi boot like we do for DS eventually).

File: src/DSi_SD.cpp [hunk 1]
```diff
@@ -973,7 +976,8 @@ void DSi_MMCStorage::SendCMD(u8 cmd, u32 param)
             RWAddress <<= 9;
             BlockSize = 512;
         }
-        RWCommand = 25;
+        if (cmd == 25)
+                RWCommand = 25;
```
Comment (RSDuck) [inline src/DSi_SD.cpp:980]:
```suggestion
    RWCommand = 25;
```

File: src/DSi_SD.cpp [hunk 2]
```diff
@@ -959,12 +960,14 @@ void DSi_MMCStorage::SendCMD(u8 cmd, u32 param)
             RWAddress <<= 9;
             BlockSize = 512;
         }
-        RWCommand = 18;
+        if (cmd == 18)
+                RWCommand = 18;
```
Comment (RSDuck) [inline src/DSi_SD.cpp:964]:
```suggestion
    RWCommand = 18;
```

**Review by RSDuck — COMMENTED** [review]
just the spacing

**Comment by asiekierka** [comment]
Solved!

**Pull request merged**